contracts/Tokens/Prime/Prime.sol

```diff
@@ -251,11 +259,14 @@ contract Prime is AccessControlledV8, PausableUpgradeable, MaxLoopsLimitHelper,
      * @custom:error Throw UserHasNoPrimeToken if user has no prime token
      * @custom:event Emits UserScoreUpdated event
      */
-    function updateScores(address[] memory users) external {
+    function updateScores(address[] calldata users) external {
         if (pendingScoreUpdates == 0) revert NoScoreUpdatesRequired();
         if (nextScoreUpdateRoundId == 0) revert NoScoreUpdatesRequired();
 
-        for (uint256 i = 0; i < users.length; ) {
+        uint256 usersLength = users.length;
+        _ensureMaxLoops(usersLength);
```

Comment:
What is the benefit of this check? Saving gas if the transaction is going to be reverted due to out-of-gas? I think I would add this check and I would leave the transaction to fail (I think it should/will be detected by the client/simulation invoking the function)